### ISSUE TITLE
Profile signaling

### ISSUE THREAD

**Issue by palemieux** [rbouqueau/TTML_in_MP4_DASH_statement#9]
Should the guidelines include recommendations on signaling the TTML profiles to which the TTML document conform?

See for instance https://www.w3.org/TR/ttml-imsc1/#profile-signaling .

**Comment by cconcolato** [comment]
Yes they should. Could you propose some text?

**Comment by palemieux** [comment]
Will do.

**Comment by palemieux** [comment]
It looks like I do not have permissions to push an issue branch and create a PR. Below is the proposed text.

---

### Determining the profile(s) of a TTML document

The following algorithm uses in-band profile signaling mechanisms to determine if the TTML document conforms to one or more common profiles. Simultaneous conformance to multiple profiles is possible since some profiles are subset of others, e.g. EBU-TT-D is a practical subset of IMSC Text Profile. Additional means of determining the profile(s) of a TTML document exist, such as explicit signaling by the user or heuristic-based inspection of the document.
- the document conforms to [IMSC1 Text Profile](https://www.w3.org/TR/ttml-imsc1/#text-profile) if the `ttp:profile` attribute of the `head` element or an `ebuttm:conformsToStandard` element is equal to `http://www.w3.org/ns/ttml/profile/imsc1/text`
- the document conforms to [IMSC1 Image Profile](https://www.w3.org/TR/ttml-imsc1/#image-profile) if the `ttp:profile` attribute of the `head` element or an `ebuttm:conformsToStandard` element is equal to `http://www.w3.org/ns/ttml/profile/imsc1/image`
- the document conforms to [EBU-TT-D] if an `ebuttm:conformsToStandard` element is equal to `urn:ebu:tt:distribution:2014-01`
- the document conforms to [SDP-US](https://www.w3.org/TR/ttml10-sdp-us) if the head element of a document contains a `ttp:profile` element with a use attribute equal to `http://www.w3.org/ns/ttml/profile/sdp-us`
- the document conforms to CFF-TT Text Profile ([UVVU DMedia](http://uvcentral.com/page/ultraviolet-online-guide-homepage)) if the `head` element contains a `ttp:profile` element with a `use` attribute of the form `http://www.decellc.org/profile/cff-tt-text-{DMEDIA_VERSION_POINTS}`, where {DMEDIA_VERSION_POINTS} is specified in [UVVU DMedia]
- the document conforms to CFF-TT Image Profile ([UVVU DMedia](http://uvcentral.com/page/ultraviolet-online-guide-homepage)) if the `head` element contains a `ttp:profile` element with a `use` attribute of the form `http://www.decellc.org/profile/cff-tt-image-{DMEDIA_VERSION_POINTS}`, where {DMEDIA_VERSION_POINTS} is specified in [UVVU DMedia]

The `ebuttm:conformsToStandard` element is specified in [EBU-TT-D].

**Comment by andreastai** [comment]
Makes sense to me. You may add that the following XSLT 1 script simply detects the profile and outputs the profile identifier. 

https://github.com/IRT-Open-Source/scf/blob/master/modules/TT-Helper/GetTTMLProfile/GetTTMLProfile.xsl

At the moment only one profile identifier is detected if one document is both, an IMSC 1 and an EBU-TT-D document, because EBU-TT-D is implictly (nearly) always a IMSC 1 document.

**Comment by rbouqueau** [comment]
Thank you both for the inputs. I think determining the TTML profile is very useful. However I think that 1) it is not related to MP4 or DASH (correct me if I'm wrong) and 2) the current document is already too long.

Could we consider putting the information in a separate document and putting a link in the original doc in case people are interested?

> It looks like I do not have permissions to push an issue branch and create a PR.

Correct. You don't have the right to create the branch on this repo but you can fork the project on github and then make a PR.
